OpenRA.Game/Scripting/ScriptContext.cs

```diff
@@ -86,8 +86,8 @@ public abstract class ScriptGlobal : ScriptObjectWrapper
 		public ScriptGlobal(ScriptContext context)
 			: base(context)
 		{
-			// The 'this.' resolves the actual (subclass) type
-			var type = this.GetType();
+			// GetType resolves the actual (subclass) type
+			var type = GetType();
```

Comment:
Are you sure you're not getting mixed up with `typeof(ScriptGlobal)` versus `this.GetType()`?